demo/src/main/java/com/example/demo/SecurityConfig.java

```diff
@@ -1,45 +1,47 @@
 package com.example.demo;
 
 
+import com.example.demo.service.CustomOAuth2UserService;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-import org.springframework.security.authentication.AuthenticationManager;
-import org.springframework.security.config.annotation.authentication.configuration.AuthenticationConfiguration;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
-import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
-import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.security.web.SecurityFilterChain;
 
-
 @Configuration
 @EnableWebSecurity
 public class SecurityConfig {
 
+    @Autowired
+    CustomOAuth2UserService customOAuth2UserService;
```

Comment:
```suggestion
@RequiredArgsConstructor
public class SecurityConfig {
    private final CustomOAuth2UserService customOAuth2UserService;
```
Autowired는 일반적으로 지양하고, 생성자 주입이 가능하면 생성자 주입을 하는 편입니다 ㅎㅎ
이는 Autowired가 지나치게 Spring Boot 의존적이고, 테스트 코드 작성에 불리함이 있기 때문입니다!